test/erlfmt_SUITE.erl

```diff
@@ -1106,3 +1135,95 @@ contains_pragma(Config) when is_list(Config) ->
             "-module(pragma).\n"
         )
     ).
+
+insert_pragma(Config) when is_list(Config) ->
+    ?assertEqual(
+            "%% @format\n"
+            "\n"
+            "-module(pragma).\n",
+        insert_pragma_string(
+            "-module(pragma).\n"
+        )
+    ),
+    ?assertEqual(
+            "%% @format\n"
+            "\n"
+            "-module(pragma).\n"
+            "\n"
+            "-export([f/3]).\n"
+            "\n"
+            "f(_Arg1, _Arg2, _Arg3) ->\n"
+            "    ok.\n",
+        insert_pragma_string(
+            "-module(pragma).\n"
+            "\n"
+            "-export([f/3]).\n"
+            "\n"
+            "f(_Arg1,_Arg2,   _Arg3) ->\n"
+            "ok.\n"
+        )
+    ),
+    ?assertEqual(
+            "%% attached comment\n"
+            "%% @format\n"
+            "-module(pragma).\n"
+            "\n"
+            "-export([f/3]).\n",
+        insert_pragma_string(
+            "%% attached comment\n"
+            "-module(pragma)\n."
+            "\n"
+            "-export([f/3]).\n"
+        )
+    ),
+    ?assertEqual(
+            "%% single comment\n"
+            "%% @format\n"
+            "\n"
+            "-module(pragma).\n"
+            "\n"
+            "-export([f/3]).\n",
+        insert_pragma_string(
+            "%% single comment\n"
+            "\n"
+            "-module(pragma)\n."
+            "\n"
+            "-export([f/3]).\n"
+        )
+    ),
+    ?assertEqual(
+            "%% LICENSE\n"
+            "%% LICENSE\n"
+            "%% LICENSE\n"
+            "%% LICENSE\n"
+            "%% @format\n"
+            "\n"
+            "-module(pragma).\n"
+            "\n"
+            "-export([f/3]).\n",
+        insert_pragma_string(
+            "%% LICENSE\n"
+            "%% LICENSE\n"
+            "%% LICENSE\n"
+            "%% LICENSE\n"
+            "\n"
+            "-module(pragma)\n."
+            "\n"
+            "-export([f/3]).\n"
+        )
+    ).
+
+insert_pragma_string(String) ->
+    StringWithPragma = format_string(String, insert),
+    %% check that insert_pragma_nodes doesn't insert a pragma, when one has already been inserted.
+    ?assertEqual(StringWithPragma, format_string(StringWithPragma, insert)),
+    StringWithPragma.
+
+format_string(String, Pragma) ->
+    {ok, Nodes, []} = erlfmt:read_nodes_string("nofile", String),
+    NodesWithPragma = case Pragma of
+        insert -> erlfmt:insert_pragma_nodes(Nodes);
+        _ -> Nodes
+    end,
+    [$\n | Formatted] = erlfmt:format_nodes(NodesWithPragma),
+    unicode:characters_to_list(Formatted).
```

Comment:
I'm not convinced about these unit tests. What do you think about adding a `erlfmt:format_string` function that would allow us to avoid exposing the two internal `erlfmt:insert_pragma_nodes` and `erlfmt:format_nodes` functions? It should allow to avoid increasing the public API area and simplifying other tests in turn as well where we're repeating a similar pattern.

Reply:
I have moved `erlfmt:format_string` into erlfmt and removed the exports for `erlfmt:insert_pragma_nodes` and `erlfmt:format_nodes`